modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/messaging/SynchronizeOccurrenceStatusListener.java

```diff
@@ -0,0 +1,39 @@
+package com.liferay.powwow.meetings.messaging;
+
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.messaging.BaseMessageListener;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.powwow.model.PowwowMeetingOccurrence;
+import com.liferay.powwow.occurrence.OccurrenceStatus;
+import com.liferay.powwow.service.PowwowMeetingOccurrenceLocalServiceUtil;
+import java.util.List;
+
+public class SynchronizeOccurrenceStatusListener extends BaseMessageListener{
+
+	@Override
+	protected void doReceive(Message message) throws Exception {
+		updateOccurrenceStatus();
+	}
+
+	private void updateOccurrenceStatus() {
+
+		List<PowwowMeetingOccurrence> powwowMeetingOccurrences =
+			PowwowMeetingOccurrenceLocalServiceUtil
+				.getPowwowMeetingOccurrences(QueryUtil.ALL_POS, QueryUtil.ALL_POS);
```

Comment:
it's recommended not fetch all data, because when the number of records is large enough, it will impact the performance.
Let's rework as below:
- have a finder method, which accepts `OccurrenceStatus` and `maxEndTime`.
- expose a service method that fetch occurrences that match status `available` and `endTime >= maxEndTime`.
- have a counting method.
- do fetching with pagination.
- update status of each occrrence.

Please see my latest finder method and service method for reference.